tools/oeedger8r/src/Sources.ml

```diff
@@ -605,12 +605,12 @@ let get_ecall_function get_deepcopy (tf : trusted_func) =
     "";
     (* Buffer validation *)
     "    /* Make sure input and output buffers lie within the enclave. */";
-    "    if (!input_buffer || !oe_is_within_enclave(input_buffer, \
-     input_buffer_size))";
-    "        goto done;";
-    "";
     "    if (!output_buffer || !oe_is_within_enclave(output_buffer, \
      output_buffer_size))";
+    "        return;";
```

Comment:
While this is a perfectly good approach to the solution, can you see if we can preserve the `goto done` approach and yet be safe. So far we have consistently tried to have a single point of return for most of our code. There may be some exceptions and glaring omissions. But that has been the intent so far.

Also keep in mind that the macros used by edger8r (`OE_ADD_SIZE`, `OE_SET_IN_POINTER` etc) will try to `goto done` in case of error and might circumvent an early return.

Reply:
Check you check again?